lib/dm-migrations/adapters/dm-do-adapter.rb

```diff
@@ -197,8 +197,14 @@ def create_unique_index_statements(model)
         def property_schema_hash(property)
           dump_class = property.dump_class
           type_map   = self.class.type_map
+          next_property_class = property.class
+          type_by_property_class = nil
+          while (type_by_property_class.nil? && next_property_class < DataMapper::Property) do
```

Comment:
Alternatively, could just look at a single parent class, rather than handling the case of the full ancestry.